### ISSUE TITLE
[SUGGESTION] 

### ISSUE THREAD

**Issue by Qwertypop4** [pokeclicker/pokeclicker#1183]
The addition of HMs and variants of fishing rods to make progression more logical

**Current Behavior**
Currently, the game includes some requirements that are usually behind HMs, as well as some spawns that would be behind HMs or fishing rods in the core series

**Improved Behavior**
I would like to add HMs such as Surf, Dive and Headbutt to affect which Pokémon are available on a given route, as well as some HMs required to access certain places such as Surf, Dive, Strength, Rock Climb and Flash.
Also, Fishing Rods would be added to affect which Pokémon spawns in certain areas.
The Old Rod from given in Kanto would be permanent, while HMs and Good/Super Rod, would depend on region.

**Reason for change**
More true to core series, less weird Pokémon spawns (for example, Sharpedo on route 103 as soon as you reach Hoenn, should require Super Rod).
Would give reason to return to earlier routes.

**Detailed Description**
I plan to add more selectors to which Pokémon can appear on a route, such as Old Rod, Good Rod, Super Rod, Surf and Headbutt. Headbutt selector already exists but would be given a use.

**Comment by KrushGames** [comment]
coming from discord, i think those are a good start to the list! since my idea for a 'flying' class got shot down before, maybe fly can be added to it. i think having gym leaders get a new purpose, like having to defeat them 10 times for an HM, is good so that people have a small incentive to do them again.

**Comment by Qwertypop4** [comment]
Here I will list my plans for the acquisition methods for each rod/HM by region and the reasons for the requirements, based on the core series games

 **Kanto**
Old Rod - complete Route 6 - given in Vermilion City
Good Rod - complete Route 15 or 18 - given in Fuchsia City
Super Rod - complete Route 12 - given in Route 12  _I may either add an extra requirement for Super Rod, for example beat Cinnabar Gym, or switch conditions for Good and Super Rod_

Cut - complete Route 6 - given in SS Anne
Surf - enter Safari Zone (if possible) or defeat Fuschia City Gym - given in Safari Zone
Fly - complete Route 16 - given in Route 16
Strength - enter Safari Zone (if possible) or defeat Fuschia City Gym - given in Fuschia City in exchange for item found in Safari Zone
Flash - complete Diglett's Cave - found in area of  Route 2 that requires traversal through Diglett's Cave (or using Cut but we'll ignore that)


**Johto**
Old Rod - complete Route 32 - given in Route 32
Good Rod - complete Route 39 - given in Olivine City
Super Rod - defeat Champion Lance - given in Route 12, which requires Champion to be beaten to reach in GSC

Cut - complete Ilex Forest - given in Ilex Forest
Headbutt - complete Ilex Forest - given in Ilex Forest
Fly - complete Route 41 - given in Cianwood City
Surf - complete Route 37 - given in Ecruteak City
Strength - complete Route 39 - given in Olivine City
Flash - complete Sprout Tower - given in Sprout Tower
Whirlpool - complete Route 42 - given in Mahogany Town
Waterfall - complete Ice Cave - found in Ice Cave



**Hoenn**
Old Rod - defeat Rustboro Gym - given in Dewford Town
Good Rod - complete Route 118 - given in Route 118
Super Rod - complete Route 125 - given in Mossdeep City

Cut - complete Route 104 - given in Rustboro City
Fly - complete Route 119 - given in Route 119
Surf - complete Petalburg Gym - given in Petalburg City after defeating Norman
Strength - complete Rusturf Tunnel - given in Rusturf Tunnel
Flash - complete Granite Cave - given in Granite Cave
Rock Smash - complete Route 110 - given in Mauville City
Waterfall - complete Sky Tower - given after calming down Kyogre and Groudon
Dive - defeat Mossdeep City Gym - given after defeating Mossdeep Gym



**Sinnoh**
Old Rod - complete Route 218 - given in the route gate to Route 218, gives reason to go to this route earlier
Good Rod - complete Route 209 - given in Route 209
Super Rod - defeat Champion Cynthia - given in Fight Area

Flash - complete Oreburgh Gate - given in Oreburgh Gate
Cut - complete Route 205 - given in Eterna City
Fly - complete Route 215 - given in Veilstone City
Surf - complete Route 210 - given in Celestic Ruins (would be altered to require completing Celestic Ruins if it was added)
Strength - complete Iron Island - given outside of Iron Island and would give reason to complete it
Defog - complete Route 209 - given in Solaceon Ruins
Rock Smash - complete Oreburgh Gate - given in Oreburgh Gate
Waterfall - defeat Sunyshore City Gym - given in Sunyshore City after defeating Volkner
Rock Climb - complete Route 217 - given in Route 217


**Unova**
Super Rod - complete Route 1 - given in Nuvema Town in BW 2

Flash - complete Burbank City Gym - given in Castelia City
Cut - defeat Aspertia City Gym - given in Virbank City
Fly - complete Route 5 - found in Route 5
Surf - complete Route 6 - found in Route 6
Strength - complete Castelia Sewers - found in Castelia Sewers
Waterfall - complete Victory Road - found in Victory Road
Dive - complete Reversal Mountain - found in Undella Town


**Kalos**
Old Rod - complete Route 8 - given in Ambrette Town
Good Rod - complete Route 12 - given in Coumarine City
Super Rod - complete Route 16 - given in Route 16

Cut - complete Parfum Palace - given in Parfum Palace
Fly - complete Route 12 - given in Coumarine City
Surf - defeat Shalour Gym - given in Shalour City after defeating Korrina
Strength - complete Connecting Cave - given in Cyllage City
Waterfall - complete Route 19 - given in Route 19


**Alola**
Fishing Rod - complete Brooklet Hill - given after trial is completed

Tauros Charge - defeat Melemele Grand Trial - given after Melemele Grand Trial
Stoutland Search - complete Paniola Ranch - given in Paniola Ranch
Lapras Paddle - complete Route 5 - given before Brooklet Hill
Charizard Glide - complete Wela Volcano Park - given after completing Fire Trial
Mudsdale Gallop - complete Route 12 - given in Route 12
Sharpedo Jet - complete Route 15 and Aether House - given on Route 15 after completing Aether House
Machamp Shove - complete Ancient Poni Path - given on Ancient Poni Path
Mantine Surf - complete Melemele Grand Trial - given after completing Melemele Grand Trial


**Galar**
Fishing Rod - no requirement - in the bag when it is obtained (might just not include this one and have it use the Alola Rod)

Rotom Bike - complete Route 9 - given on Route 9

Some of these (most likely Fly) may not be added due to lack of use.

For Flash and Defog, I would like to make the areas affected more difficult when they are not obtained, but will probably be incapable of this

**Comment by Ultima1990** [comment]
I'd say mainly stick to rods and surf for this. None of the other mechanics really lock pokémon encounters, just areas. By which I mean yes, surf also locks areas, but you also encounter pokémon _while_ surfing. You don't really encounter pokémon _while_ using strength, or cut, or rock climb, etc. Yes you encounter pokémon _while_ using flash, but that's by no means required. Except Headbutt. Which, I think, is planned to be an ability in the party? Maybe rock smash can be considered, though it doesn't really lock unique encounters. I do like having Dive on this list. That one makes sense to me.

For route progression the obtaining of HMs is implied by the way the routes and dungeons unlock. I don't think having the HMs visible in some menu somewhere adds much for this purpose.

**Comment by Qwertypop4** [comment]
I agree with that, most were just there in case they were wanted. Just Fishing Rods, Surf/Rotom Bike, Dive and Headbutt lock spawns behind them.so they are the only ones that would be required, if this were to be implemented
